src/notebook/notebook/modelfactory.ts

```diff
@@ -18,9 +18,9 @@ import {
  * A model factory for notebooks.
  */
 export
-class NotebookModelFactory implements IModelFactory {
+class NotebookModelFactory extends TextModelFactory implements IModelFactory {
```

Comment:
What's the reason behind making this a subclass? It seems to be a different enough thing that a notebook model is not a text model.